backend/rag_optimizers.py

```
@@ -89,45 +96,45 @@ def rerank_results(query: str, chunks: List[Dict[str, Any]], reranker_model_name
     except ImportError:
-        print("[RAG OPTIMIZER] sentence-transformers not installed. Skipping re-ranking.")
+        logger.warning("sentence-transformers not installed. Skipping re-ranking.")
         return chunks
 
     if reranker_model_name not in _RERANKER_CACHE:
-        print(f"[RAG OPTIMIZER] Loading Cross-Encoder: {reranker_model_name}...")
+        logger.info("Loading Cross-Encoder: %s", reranker_model_name)
         try:
             # Load locally or download automatically
             _RERANKER_CACHE[reranker_model_name] = CrossEncoder(reranker_model_name)
         except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)